src/AccessInput.php

```diff
@@ -42,19 +42,33 @@ public function run()
             ->textInput(['readonly' => true]); // TODO: Check owner in model (has to be the same as current user)
 
         foreach (['domain', 'read', 'update', 'delete'] as $access) {
+            // save the current value in disabled because we may rewrite it if value is not in data
+            $originalDisabled = $disabled;
+
+            $data = $access === 'domain' ? $userDomains : $userAuthItems;
             $fieldName = 'field' . ucfirst($access);
-            $return .= $this->form->field($this->model, $this->{$fieldName})->widget(
-                Select2::classname(),
+
+            $vaule = $this->model->{$this->$fieldName};
+            // Check if value set is in data list and add it if its not and disable the input
+            if (!array_key_exists($vaule, $data)) {
+                $data[$vaule] = $vaule;
+                $disabled = !Yii::$app->getUser()->can($vaule); // Check if current user is able to modify the value
```

Comment:
By checking if the current user has this role in some way assigned to him it allows to give users the opportunity to edit the value.

Reply:
Needed for special cases when a more privileged user is able to do anything. Looking at you `yii\dmstr\User`